### PR TITLE
[review] Create a 'tmpfile' library for tests only

### PR DESCRIPTION
**Pull request by dawagner**
Also, this removes TmpFile from pfw_utility which in turn was part of
libparameter. TmpFile uses unsecure functions for creating temporary files.
While this is not an issue for test purposes, it can be a security issue
elsewhere.

Even though TmpFile was only used for test purposes, having it part of
libparameter raises security warnings in static code analysers; removing it
from libparameter removes any chance of using it.

Signed-off-by: David Wagner <david.wagner@intel.com>
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/302%23issuecomment-152579949%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-30T16%3A34%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2087aa842debdf0e474f165e28b6a6deefa8199869%20test/toolbox/CMakeLists.txt%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/302%23discussion_r43499002%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20disagree%20merging%20catch%20with%20test-utility%20%28pfw-test-toolbox%20feel%20like%20a%20catch-all%20lib%29%3A%5Cr%5Cn%20-%20one%20reflects%20an%20external%20library%5Cr%5Cn%20-%20the%20other%20is%20a%20custom%20class%22%2C%20%22created_at%22%3A%20%222015-10-30T13%3A00%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20done%22%2C%20%22created_at%22%3A%20%222015-10-30T14%3A59%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/toolbox/CMakeLists.txt%3AL46-59%22%7D%2C%20%22Pull%2087aa842debdf0e474f165e28b6a6deefa8199869%20bindings/c/CMakeLists.txt%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/302%23discussion_r43497760%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22pfw_utility%20%3D%3E%20test-utility%20%3F%20%28why%20is%20it%20prefixed%20by%20%60pfw-%60%20anyway%20%28it%20is%20not%20a%20public%20lib%29%29%22%2C%20%22created_at%22%3A%20%222015-10-30T12%3A43%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20renamed%20it%20tmpfile%22%2C%20%22created_at%22%3A%20%222015-10-30T14%3A59%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/CMakeLists.txt%3AL49-56%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/302%23issuecomment-152579949%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/302%23discussion_r43497760%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/302%23discussion_r43499002%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/302%23discussion_r43511181%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/302%23discussion_r43511194%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [x] <a href='#crh-comment-Pull 87aa842debdf0e474f165e28b6a6deefa8199869 bindings/c/CMakeLists.txt 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/302#discussion_r43497760'>File: bindings/c/CMakeLists.txt:L49-56</a></b>
- <a href='https://github.com/krocard'></a> pfw_utility => test-utility ? (why is it prefixed by `pfw-` anyway (it is not a public lib))
- <a href='https://github.com/dawagner'></a> :+1: renamed it tmpfile
- [x] <a href='#crh-comment-Pull 87aa842debdf0e474f165e28b6a6deefa8199869 test/toolbox/CMakeLists.txt 14'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/302#discussion_r43499002'>File: test/toolbox/CMakeLists.txt:L46-59</a></b>
- <a href='https://github.com/krocard'></a> I disagree merging catch with test-utility (pfw-test-toolbox feel like a catch-all lib):
- one reflects an external library
- the other is a custom class
- <a href='https://github.com/dawagner'></a> :+1: done

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/302?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/302?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/302?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/302'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/302'>Refresh</a>.<a href='#crh-update-none'></a>

File: bindings/c/CMakeLists.txt [hunk 1]
```diff
@@ -49,7 +49,7 @@ if(BUILD_TESTING)
     # Add unit test
     add_executable(cparameterUnitTest Test.cpp)
 
-    target_link_libraries(cparameterUnitTest PRIVATE catch cparameter pfw_utility)
+    target_link_libraries(cparameterUnitTest PRIVATE cparameter pfw-test-toolbox)
```
Comment (krocard) [inline bindings/c/CMakeLists.txt]: pfw_utility => test-utility ? (why is it prefixed by `pfw-` anyway (it is not a public lib))

File: test/toolbox/CMakeLists.txt [hunk 2]
```diff
@@ -46,6 +46,13 @@ if(BUILD_TESTING)
...
-    target_include_directories(catch PUBLIC "${CATCH_HEADER}")
+    if (WIN32)
+        set(OS_SPECIFIC_TMPFILE windows/TmpFile.cpp)
+    else ()
+        set(OS_SPECIFIC_TMPFILE posix/TmpFile.cpp)
+    endif ()
+
+    add_library(pfw-test-toolbox STATIC catch.cpp ${OS_SPECIFIC_TMPFILE})
+    target_include_directories(pfw-test-toolbox
+        PUBLIC . "${CATCH_HEADER}")
```
Comment (krocard) [inline test/toolbox/CMakeLists.txt]: I disagree merging catch with test-utility:
 - one reflects an external library
 - one 
+ i dislikes junk lib

File: bindings/c/CMakeLists.txt (hunk 1 above)
Comment (dawagner) [inline bindings/c/CMakeLists.txt]: :+1: renamed it tmpfile

File: test/toolbox/CMakeLists.txt (hunk 2 above)
Comment (dawagner) [inline test/toolbox/CMakeLists.txt]: :+1: done

**Comment by krocard** [comment]
:shipit:

**Pull request merged**